src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/OtlpExporterPersistentStorageRetryTransmissionHandler.cs

```diff
@@ -0,0 +1,110 @@
+using OpenTelemetry.PersistentStorage.Abstractions;
+using OpenTelemetry.PersistentStorage.FileSystem;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal sealed class OtlpExporterPersistentStorageRetryTransmissionHandler<TRequest> : OtlpExporterTransmissionHandler<TRequest>
+{
+    private readonly ManualResetEvent stopEvent = new(false);
+    private readonly Thread thread;
+    private readonly PersistentBlobProvider? persistentBlobProvider;
+    private readonly Func<byte[], TRequest>? requestFactory;
+
+    public OtlpExporterPersistentStorageRetryTransmissionHandler(IExportClient<TRequest> exportClient, Func<byte[], TRequest> requestFactory, string storagePath)
+        : base(exportClient)
+    {
+        this.persistentBlobProvider = new FileBlobProvider(storagePath);
+        this.requestFactory = requestFactory;
+
+        this.thread = new Thread(this.ProcessRetryRequestsThreadBody)
+        {
+            Name = $"OtlpExporter Persistent Retry Storage - {typeof(TRequest)}",
+            IsBackground = true,
+        };
+
+        this.thread.Start();
+    }
+
+    protected override bool OnSubmitRequestFailure(TRequest request, ExportClientResponse response)
+    {
+        byte[]? data = null;
+        if (request is ExportTraceServiceRequest traceRequest)
+        {
+            data = traceRequest.ToByteArray();
+        }
+        else if (request is ExportMetricsServiceRequest metricsRequest)
+        {
+            data = metricsRequest.ToByteArray();
+        }
+        else if (request is ExportLogsServiceRequest logsRequest)
+        {
+            data = logsRequest.ToByteArray();
+        }
+
+        if (data != null)
+        {
+            this.persistentBlobProvider?.TryCreateBlob(data, out _);
+        }
+
+        return true;
+    }
+
+    protected override void OnShutdown()
+    {
+        this.stopEvent.Set();
+        this.thread.Join();
+        base.OnShutdown();
+    }
+
+    private void ProcessRetryRequestsThreadBody()
+    {
+        while (true)
+        {
+            // Wait before retrying
+            if (this.stopEvent.WaitOne(5000))
+            {
+                break;
+            }
+
+            int fileCount = 0;
+
+            // transmit 10 files at a time.
+            while (fileCount < 10)
+            {
+                if (this.persistentBlobProvider != null && this.persistentBlobProvider.TryGetBlob(out var blob))
+                {
+                    if (blob != null && blob.TryLease(20000) && blob.TryRead(out var data))
+                    {
+                        if (this.requestFactory != null)
+                        {
+                            var request = this.requestFactory.Invoke(data);
+                            var response = this.RetryRequest(request);
```

Comment:
No call into `ShouldRetryRequest` here either? Wouldn't we want to drop stuff after a certain time period or enough retry attempts?

Reply:
Just now realizing you switched it so this doesn't derive from `OtlpExporterRetryTransmissionHandler` interesting. I guess I was assuming retry rules would apply to persistent version and in-memory version.

Reply:
We likely dont need the ShouldRetryRequest here, if the transmission fails it will be kept in storage for retrying after the delay. Telemetry that is stored offline will be cleaned up by PersistentStorageProvider background thread after a configured time period(default 2 days). (so it will be retried periodically during that time).

Reply:
Could you clarify what you mean by retry rules?

Reply:
What I thought would happen is each time we go to retry something we would call `ShouldRetryRequest` passing a new attempt count and then the rule engine would decided if we have tried enough, or has the right conditions for retry, or whatever. May not be necessary just wondering.

Reply:
While trying to transmit from offline storage, we do not need to manage retryCount. But re-thinking about it, there is a need to check ShouldRetry here. For e.g. the telemetry was stored offline because the server was unavailable and now we retry it again and get 400-Bad request. In this scenario we should remove the telemetry from storage.